### ISSUE TITLE
[All] Clear serialization language flag 

### ISSUE THREAD

**Issue by LiangliangSui** [apache/incubator-fury#1607]
## Is your feature request related to a problem? Please describe.
The current Fury implementation (Java/golang/javascript, etc.), the language used for serialization will be added to the Fury Header(Contradicts the current xlang_spec).

Fury Header looks like the following format.

```
|    2 bytes   |     4 bits    | 1 bit | 1 bit | 1 bit  | 1 bit |  1 byte  |          optional 4 bytes              |
+--------------+---------------+-------+-------+--------+-------+----------+-------------------------------------+
| magic number | reserved bits |  oob  | xlang | endian |  null | language | unsigned int for meta start offset |
```

code show as below

![image](https://github.com/apache/incubator-fury/assets/116876207/2e2c63b5-d857-41bf-af8a-b679ef484858)

![image](https://github.com/apache/incubator-fury/assets/116876207/9c7d6b06-5ab7-4915-9dc1-c4eee5b6639b)

## Describe the solution you'd like
Do we need to remove these redundant language identifiers?

**Comment by LiangliangSui** [comment]
cc @chaokunyang

**Comment by chaokunyang** [comment]
If the deserialization knows the encoding language, it may be able to do better encoding. For example, if sender is golang, and receiver is javascript, golang can use utf16 as the encoding instead of native utf8, then javascript deserialization will be faster.

**Comment by chaokunyang** [comment]
I think we can keep this currently, and add the design choice to the spec

**Comment by LiangliangSui** [comment]
Yes, we can also keep it and add an explanation in xlang_spec. This will create an extra byte in the Fury Header. I'll update the spec later
